xml/xml-parse.lisp

```diff
@@ -2617,7 +2617,9 @@
               (let ((xi2 (xstream-open-extid effective-extid)))
 		(with-zstream (zi2 :input-stack (list xi2))
 		  (ensure-dtd)
-                  (sax:start-internal-subset (handler *ctx*))
+                  ;; Avoid duplicate internal subsets.
+                  (unless (ignore-errors (have-internal-subset (handler *ctx*)))
```

Comment:
Why is the `ignore-errors` necessary? The only method on `have-internal-subset` is a slot reader, the slot has `:initform nil` and there doesn't seem to be a `slot-makunbound' call for the slot.

Reply:
Just running the tests, I see that `cxml:sax-proxy` does not inherit from `cxml:sink`, and so does not have an `internal-subset` slot at all, and `rune-dom::dom-builder`, while it does inherit from `cxml:sink`, overrides the definition of the slot and does not provide an initform. Of course there might be others.